Applications/Utils/MeshEdit/UnityPreprocessing.cpp

```diff
@@ -0,0 +1,245 @@
+/*
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include <string>
+#include <vector>
+
+#include <tclap/CmdLine.h>
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/MeshInformation.h"
+#include "MeshLib/Elements/Line.h"
+#include "MeshLib/Elements/Tri.h"
+#include "MeshLib/Elements/Quad.h"
+#include "MeshLib/Elements/Tet.h"
+#include "MeshLib/Elements/Hex.h"
+#include "MeshLib/Elements/Pyramid.h"
+#include "MeshLib/Elements/Prism.h"
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "MeshLib/IO/VtkIO/VtuInterface.h"
+#include "MeshLib/MeshSearch/ElementSearch.h"
+#include "MeshLib/MeshEditing/RemoveMeshComponents.h"
+
+template <class T>
+bool checkType(MeshLib::Properties const& props, std::string const& name)
+{
+    if (props.existsPropertyVector<T>(name))
+    {
+        MeshLib::PropertyVector<T> const*const vec = props.getPropertyVector<T>(name);
+        if (vec->getMeshItemType() == MeshLib::MeshItemType::Cell)
+            return true;
+    }
+    return false;
+}
+
+bool isCellVec(MeshLib::Properties const& props, std::string const& name)
+{
+    if (checkType<int>(props, name))
+        return true;
+    if (checkType<double>(props, name))
+        return true;
+    if (checkType<long>(props, name))
+        return true;
+    return false;
+}
+
+bool containsCellVecs(MeshLib::Mesh const& mesh)
+{
+    MeshLib::Properties const& props (mesh.getProperties());
+    std::vector<std::string> const& vec_names(props.getPropertyVectorNames());
+    for (std::string name : vec_names)
+    {
+        if (isCellVec(props, name))
+            return true;
+    }
+    return false;
+}
+
+template <class T>
+MeshLib::Element* createElement(MeshLib::Element const& e,
+    std::vector<MeshLib::Node*> &nodes,
+    std::vector<std::vector<std::size_t>> &node_map)
+{
+    std::size_t const n_nodes = e.getNumberOfNodes();
+    MeshLib::Node** new_nodes = new MeshLib::Node*[n_nodes];
```

Comment:
This isn't a mem leak. The new_nodes pointer is deleted in the destructor for the Element-class.